src/geo/transform.js

```diff
@@ -1969,6 +1980,29 @@ class Transform {
         return this.scaleZoom(this.cameraToCenterDistance / (z * this.tileSize));
     }
 
+    // This function is helpful to approximate true zoom given a mercator height with varying ppm.
+    // With Globe, since we use a fixed reference latitude at lower zoom levels and transition between this
+    // latitude and the center's latitude as you zoom in, camera to center distance varies dynamically.
+    // As the cameraToCenterDistance is a function of zoom, we need to approximate the true zoom
+    // given a mercator meter value in order to eliminate the zoom/cameraToCenterDistance dependency.
+    zoomFromMercatorZAdjusted(z: number): number {
+        const getZoom = (zoom) => {
+            const d = this.getCameraToCenterDistance(this.projection, zoom);
+            return this.scaleZoom(d / (z * this.tileSize));
+        };
+
+        let zoom = getZoom(this.zoom);
+        let diff = Math.abs(zoom - getZoom(zoom));
+        let lastdiff;
+        while (lastdiff !== diff) {
```

Comment:
It should converge in a few iterations, but as a safeguard it would be wiser to add a max iteration count or an epsilon comparison, as it's difficult to confirm that floating point comparison will not break this. This method will not be necessary in gl-native, as adaptive scaling to get constant globe size isn't applied yet.